### PR TITLE
Virtualized lists

### PR DESCRIPTION
**Pull request by musangowope**
- Created TOCList, which is a consolidated component of rulesTOC.tsx and constitutionTOC.tsx. Used TOCList in pages components: RulesTOC.tsx and ConstitutionTOC.tsx
- In TOCList implement virtualised list logic to optimize performance with package Virtuoso
- Made IonLists naturally use remaining vertical space with flex
- Implement logic to only apply virtualized list on a mobile screen width. Desktop browsers are pretty sound
- Delete RuleTOC.tsx and ConstitutionTOC.tsx and refactor other usages to use TOCList instead
- apply placeholder ux for large scrolling

Comparison videos:
Demo video with this PR Logic:
switching between tabs is faster

https://user-images.githubusercontent.com/15995210/148218719-c4513ef7-deb0-4586-84e5-af3d6ad6bd02.mp4

Demo of current constitution app on mobile browser without PR logic
I tried to switch between tabs at a semi rapid rate. the my browser crashed

https://user-images.githubusercontent.com/15995210/148218916-8b1c6ad6-0ebc-4117-bc2e-bb2f15c6f61b.mp4

### CODE REVIEW

File: src/data/toc.ts [hunk 1]
```diff
@@ -54,3 +57,21 @@ function flattenTOC(arr: any[]) {
   return result;
 }
 
+// @ts-ignore
+function flattenTOCDeep(array: any[]) {
```
Comment (longhotsummer) [inline src/data/toc.ts:61]: This existing `flattenTOC` just keeps the ids. Why not adjust that function to store the entire item (and add the depth attribute), and then re-created the old `flattened` list of ids based on the complete flattened list? Or, where is the old flattened list used, and can you adjust it to use the complete flattened toc?

Comment (musangowope) [reply src/data/toc.ts]: Done

**Comment by musangowope** [comment]
@longhotsummer I'm marking this pr as ready. Its worth chatting through it though. Lmk when you're available to chat

File: src/components/TOCList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import React from "react";
+import {Virtuoso} from 'react-virtuoso'
+import Row from "./Row";
+import {useWindowSize} from "../../custom-hooks/useWindowResize";
+
+export interface iTOCItem {
```
Comment (musangowope) [inline src/components/TOCList/index.tsx:6]: change to I

**Pull request merged**